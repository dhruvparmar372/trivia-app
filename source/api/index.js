import axios from "axios";
// import qs from "qs";
import Config from "react-native-config";

const api = axios.create({
  baseURL: Config.API_HOST
  // transformRequest: [
  //   data => {
  //     return typeof data === "object" ? qs.stringify(data) : data;
  //   }
  // ]
});

export default api;
