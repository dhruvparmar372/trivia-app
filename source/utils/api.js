import _shuffle from "lodash.shuffle";

export function parseQuestions(questions, quizId) {
  return questions.map((question, index) => ({
    ...question,
    id: `${quizId}${index}`,
    answers: _shuffle([...question.incorrect_answers, question.correct_answer])
  }));
}
