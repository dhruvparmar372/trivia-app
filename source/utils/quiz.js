export function isQuizComplete(quiz) {
  return quiz
    ? quiz.questions.reduce(
        (isComplete, question) => isComplete && !!question.answer,
        true
      )
    : false;
}

export function getFirstUnansweredQuestion(quiz) {
  return quiz
    ? quiz.questions.reduce(
        (accum, question) => accum || (question.answer ? null : question),
        null
      )
    : null;
}

export function isAnswerCorrect(question = {}, answer = {}) {
  const { value } = answer;
  const { correct_answer } = question;
  if (!value || !correct_answer) {
    return false;
  }

  switch (question.type) {
    case "boolean":
      return correct_answer === value;
    default:
      return correct_answer === value;
  }
}

export function getQuizScore(quiz) {
  if (!quiz) {
    return "";
  }

  const total = quiz.questions.length;
  const correct = quiz.questions.reduce(
    (count, question) =>
      count + (isAnswerCorrect(question, question.answer) ? 1 : 0),
    0
  );

  return {
    correct,
    total
  };
}

export function formatQuizScore(score) {
  return score.correct / score.total * 100;
}
