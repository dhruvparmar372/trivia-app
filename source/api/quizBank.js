import api from "source/api";

export const fetchQuestions = params => api.get("/", { params });
