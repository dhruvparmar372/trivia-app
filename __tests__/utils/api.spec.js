import { parseQuestions } from "source/utils/api";

const dummyQuizId = 201;
const dummyQuestions = [
  {
    question: "Question 1",
    incorrect_answers: [],
    correct_answer: "True"
  },
  {
    question: "Question 2",
    incorrect_answers: [],
    correct_answer: "True"
  }
];

test("parse questions adds id to all questions", () => {
  expect(parseQuestions(dummyQuestions, dummyQuizId)).toMatchSnapshot();
});
