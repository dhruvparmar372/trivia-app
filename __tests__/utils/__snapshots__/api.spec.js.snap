// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse questions adds id to all questions 1`] = `
Array [
  Object {
    "answers": Array [
      "True",
    ],
    "correct_answer": "True",
    "id": "2010",
    "incorrect_answers": Array [],
    "question": "Question 1",
  },
  Object {
    "answers": Array [
      "True",
    ],
    "correct_answer": "True",
    "id": "2011",
    "incorrect_answers": Array [],
    "question": "Question 2",
  },
]
`;
